universal-idl-program/program/src/instructions/extend/mod.rs

```diff
@@ -0,0 +1,114 @@
+use crate::{
+    account_checks::{
+        assert_executable, assert_mut, assert_signer, assert_system_program, assert_with_msg,
+    },
+    instructions::common::transfer,
+    state::{assert_buffer_seeds, Idl, SolanaAccount},
+    IdlProgramInstruction,
+};
+use borsh::{BorshDeserialize, BorshSerialize};
+use lazy_format::lazy_format;
+use solana_program::{account_info::next_account_info, pubkey::Pubkey, system_program};
+use solana_program::{account_info::AccountInfo, program_error::ProgramError};
+use solana_program::{entrypoint::ProgramResult, msg};
+use solana_program::{instruction::AccountMeta, rent::Rent};
+use solana_program::{instruction::Instruction, sysvar::Sysvar};
+
+pub fn extend(
```

Comment:
This instruction name really confused me, I think `write_buffer` is a lot more sensible

Reply:
it's not writing to the buffer however, it's extending.

How about `extend_buffer_data` ?